src/styles/main.scss

```
@@ -1,3 +1,68 @@
+$square-size: 100px;
+$padding: 10px;
+$gap: 1px;
+$width-content: calc((#{$square-size} + #{$gap}) * 7 + #{$padding} * 2);
+$day-of-week:
+  "mon" 0,
+  "tue" 1,
+  "wed" 2,
+  "thu" 3,
+  "fri" 4,
+  "sat" 5,
+  "sun" 6;
+
+@mixin square ($size) {
+  width: $size;
+  height: $size;
+}
+
 body {
   margin: 0;
 }
+
+html {
+  font-family: Arial, Helvetica, sans-serif;
+  font-size: 30px;
+}
+
+.calendar {
+  display: flex;
+  gap: $gap;
+  width: $width-content;
+  flex-wrap: wrap;
+  margin: 50px auto;
+  padding: $padding;
+
+  @each $day, $num in $day-of-week {
+    &--start-day--#{$day} :first-child {
+      margin-left: calc((#{$square-size} + #{$gap}) * #{$num});
+    }
+  }
+
+  @for $len from 28 through 31 {
+    &--month-lenth--#{$len} :nth-child(n + #{$len + 1}) {
+      display: none;
+    }
+  }
+}
+
+.calendar__day {
+  display: flex;
+  @include square ($square-size);
+  background-color: #eee;
+  border: 1px solid black;
+  box-sizing: border-box;
+  cursor: pointer;
+
+  &:hover {
+    background-color: pink;
+    transform: translateY(-20px);
+    transition: 0.5s ease-in;
```

1. If you put transition parameters under general selector (not `:hover`) it will work both ways.
2. Always add rules that you apply transition to:
```suggestion
    transition: background-color 0.5s ease-in, transform 0.5s ease-in;
```